src/Sylius/Bundle/ApiBundle/Serializer/CommandFieldItemIriToIdentifierDenormalizer.php

```diff
@@ -13,46 +13,53 @@
 
-use ApiPlatform\Core\Api\IriConverterInterface;
-use Sylius\Bundle\ApiBundle\Command\AddProductReview;
+use Sylius\Bundle\ApiBundle\Converter\ItemIriToIdentifierConverterInterface;
 use Sylius\Bundle\ApiBundle\DataTransformer\CommandAwareInputDataTransformer;
-use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Bundle\ApiBundle\Map\CommandItemIriArgumentToIdentifierMapInterface;
 use Symfony\Component\Serializer\Normalizer\ContextAwareDenormalizerInterface;
 use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
 
-/** @experimental */
-final class AddProductReviewCommandFieldItemIriToIdentifierDenormalizer implements ContextAwareDenormalizerInterface
+final class CommandFieldItemIriToIdentifierDenormalizer implements ContextAwareDenormalizerInterface
 {
     /** @var DenormalizerInterface */
     private $objectNormalizer;
 
+    /** @var ItemIriToIdentifierConverterInterface */
+    private $itemIriToIdentifierConverter;
+
     /** @var CommandAwareInputDataTransformer */
     private $commandAwareInputDataTransformer;
 
-    /** @var IriConverterInterface */
-    private $iriConverter;
+    /** @var CommandItemIriArgumentToIdentifierMapInterface */
+    private $commandItemIriArgumentToIdentifierMap;
 
     public function __construct(
         DenormalizerInterface $objectNormalizer,
+        ItemIriToIdentifierConverterInterface $itemIriToIdentifierConverter,
         CommandAwareInputDataTransformer $commandAwareInputDataTransformer,
-        IriConverterInterface $iriConverter
+        CommandItemIriArgumentToIdentifierMapInterface $commandItemIriArgumentToIdentifierMap
     ) {
         $this->objectNormalizer = $objectNormalizer;
+        $this->itemIriToIdentifierConverter = $itemIriToIdentifierConverter;
         $this->commandAwareInputDataTransformer = $commandAwareInputDataTransformer;
-        $this->iriConverter = $iriConverter;
+        $this->commandItemIriArgumentToIdentifierMap = $commandItemIriArgumentToIdentifierMap;
     }
 
     public function supportsDenormalization($data, $type, $format = null, array $context = [])
     {
-        return $this->getInputClassName($context) === AddProductReview::class ? true : false;
+        return $this->commandItemIriArgumentToIdentifierMap->has($this->getInputClassName($context));
     }
 
     public function denormalize($data, $type, $format = null, array $context = [])
     {
-        /** @var ProductInterface $product */
-        $product = $this->iriConverter->getItemFromIri($data['product']);
+        /** @psalm-var class-string $inputClassName */
+        $inputClassName = $this->getInputClassName($context);
+
+        $fieldName = $this->commandItemIriArgumentToIdentifierMap->get($inputClassName);
 
-        $data['product'] = $product->getCode();
+        if (array_key_exists($fieldName, $data)) {
+            $data[$fieldName] = $this->itemIriToIdentifierConverter->getIdentifier($data[$fieldName]);
+        }
```

Comment:
Also, improvement for the second iteration, but what is blocking us from checking if the field consists IRI and execute this logic on all fields like this? This way, we could remove the whole map from the concept